### DISCUSSION TITLE
Kesulitan melakukan filter time series data  (Capstone ML)

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#625] [Q&A] [answered]
Data yang saya miliki mengenai kasus Time Series FnB.

Pada saat ini, saya sudah berhasil melakukan :
- Pembulatan waktu
- Group by transaction date, untuk menunjukkan jumlah transaksi per jamnya
- Padding untuk waktu-waktu yang kosong
- Penggantian nilai-nilai NA yang muncul dari padding menjadi 0

Proses yang ingin saya lakukan selanjutnya adalah untuk memfilter data yang saya miliki untuk hanya menampilkan data jam berdasarkan jam buka toko FnB kasus sebelum dijadikan model time series. Saya juga sudah mengassign object yang dapat memfilter hal tersebut, kode dan hasilnya sebagai berikut:

```
dt_fltr<- train_clean$transaction_date[hour(train_clean$transaction_date) >= 10 & hour(train_clean$transaction_date) <= 22]

[1] "2017-12-01 10:00:00 UTC" "2017-12-01 11:00:00 UTC" "2017-12-01 12:00:00 UTC"
[4] "2017-12-01 13:00:00 UTC" "2017-12-01 14:00:00 UTC" "2017-12-01 15:00:00 UTC"
[7] "2017-12-01 16:00:00 UTC" "2017-12-01 17:00:00 UTC" "2017-12-01 18:00:00 UTC"
[10] "2017-12-01 19:00:00 UTC" "2017-12-01 20:00:00 UTC" "2017-12-01 21:00:00 UTC"

# dan seterusnya...
```

Namun ketika ingin diapply ke data train saya (yang bernama train_clean), muncul error sebagai berikut:

```
train_clean[dt_fltr]

Error in `train_clean[dt_fltr]`:
! Can't subset columns with `dt_fltr`.
✖ `dt_fltr` must be logical, numeric, or character, not a <POSIXct/POSIXt> object.
Backtrace:
 1. train_clean[dt_fltr]
 7. rlang::cnd_signal(x)
```
Dari apa yang saya mengerti, sepertinya package dplyr sama sekali tidak mau bekerja dengan tipe data POSIXct, POSIXt, dan bahkan Date.

Apakah ada cara lain untuk dapat melakukan filter selain menggunakan dplyr?

Terima kasih sebelumnya.

**Comment by nadyadtm** [comment, marked as answer]
Dear Pak @AGelfP 

Bapak dapat melakukan filter data dengan menggunakan `%in%`. Contoh penggunaannya adalah sebagai berikut

`train_data[train_data$transaction_date %in% dt_fltr, ]`

Semoga dapat membantu, bila masih ada kendala dapat langsung disampaikan saja. Terima kasih

**Reply by AGelfP** [reply]
Ok Kak, akhirnya berhasil juga.... Terima kasih ya Kak

**Reply by nadyadtm** [reply]
Baik pak, sama-sama
